frontend/src/views/groupmember/UpdateGroupMember.vue

```diff
@@ -0,0 +1,151 @@
+        <AppButton :submitting="submitting" mode="primary" type="submit">Submit</AppButton>
+        <AppButton @click="resetForm" mode="secondary" type="button">Clear</AppButton>
+      </AppRow>
+    </form>
+  </div>
+  <br />
+  <div id = "confirmation" v-if="updateOk">
+    <p>PHN: {{ result?.phn }}</p>  
+  </div>
+</template>
+
+<script>
+import GroupMemberService from '../../services/GroupMemberService'
+import useVuelidate from '@vuelidate/core'
+import { validateGroupNumber, validatePHN, VALIDATE_PHN_MESSAGE, VALIDATE_GROUP_NUMBER_MESSAGE, VALIDATE_OPTIONAL } from '../../util/validators'
+import { required, helpers } from '@vuelidate/validators'
+
+function validateOptional() {
+  if (this.departmentNumber !== '' || this.groupMemberNumber !== '') {
+    return true;
+  } else {
+    return false;
+  }
+}
+
+export default {
+  name: 'UpdateGroupMember',
+  setup() {
+    return {
+      v$: useVuelidate()}
+  },
+  data() {
+    return {
+      groupNumber:'',
+      phn: '',
+      groupMemberNumber: '',
+      departmentNumber: '',
+      updateOk: false, 
+      showForm : true,
+      submitting: false,
+      result: {
+        phn: '', 
+        status: '',
+        message: '',
+      }
+    }
+  },
+  
+  methods: {
+    async submitForm() {
+      this.result = null 
+      this.updateOk = false
+      this.showForm = true
+      this.$store.commit("alert/dismissAlert")
+      try {
+        const isValid = await this.v$.$validate()
+        if (!isValid) {
+          this.showError()
+          return
+        } 
+        this.result = (await GroupMemberService.updateGroupMember({
+          phn: this.phn,
+          groupNumber: this.groupNumber,
+          departmentNumber: this.departmentNumber,
+          groupMemberNumber: this.groupMemberNumber,         
+        })).data  
+        
+        if (this.result.status === 'error') {
+          this.$store.commit('alert/setErrorAlert', this.result.message)
+          return
+        }
+
+        if (this.result?.status === 'success') {
+          this.showForm = false
+          this.updateOk = true
+          this.$store.commit('alert/setSuccessAlert', this.result.message)
+          return
+        }
+      
+      } catch (err) {
+        this.$store.commit('alert/setErrorAlert', `${err}`)
+      }
+    },
+    showError(error) {
+      this.$store.commit('alert/setErrorAlert', error)
+      this.result = {}
+    },
+    resetForm() {
+      this.phn = ''
+      this.groupMemberNumber = ''
+      this.groupNumber = ''
+      this.departmentNumber = ''
+      this.result = null
+      this.updateOk = false
+      this.showForm = true
+      this.v$.$reset()
+      this.$store.commit("alert/dismissAlert") 
+    }
+  },
+  validations() {
+    return {
+      phn: {
+        required,
+        validatePHN: helpers.withMessage(
+          VALIDATE_PHN_MESSAGE, validatePHN
+        )
+      },
+      groupNumber: { required,
+      validateGroupNumber: helpers.withMessage(
```

Comment:
Formatting is inconsistent: required should be on next line and validateGroupNumber should be indented.